src/templates/DatasetSearch/index.jsx

```diff
@@ -104,6 +104,13 @@ const DatasetSearch = ({
     }
   }, [fulltext]);
 
+  useEffect(() => {
+    // update search on browser back button press
+    const updatedFullText = location.search.indexOf("=") !== -1 ? location.search.split("=")[1] : "";
```

Comment:
We need a check to make sure this is actually the fulltext search param. I clicked on page 5 without doing a search and it filled the url bar in the browser and the fulltext search input with `fulltext=5%25252525252525252525252525252525252525252525252525252525252525252525252525252525252525252525252525252525252525252525252525252525252525252525252525252525252525252525252525252525252525252525252525252525252525252525252525252525252525252525252525252525252525252525252525252525252525252525252525252525252525252525252525252525252525252525252525252525252525252525252525252525252525252525252525252526fulltext`

Reply:
Great catch! It should be isolated now. Tested clicking around a few pages and filters and it seems to function correctly now for me.